jobrunner/create_or_update_jobs.py

```diff
@@ -185,13 +185,15 @@ def recursively_add_jobs(job_request, project, action, force_run_actions):
         if required_job:
             wait_for_job_ids.append(required_job.id)
 
+    database_name = 'dummy' if config.USING_DUMMY_DATA_BACKEND else job_request.database_name
```

Comment:
I think I'd prefer to leave this as it's original value and just not validate it when using dummy data. It doesn't do any harm to leave it as is and I think the fewer places in the code we can fork on `config.USING_DUMMY_DATA_BACKEND` the better.